src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java

```diff
@@ -19,11 +21,17 @@ public String obtainBallot(String id, String adminKey) {
 
 		if (adminKey.equals(vote.adminKey))
 			vote.increaseRecordedBallots("admin");
-		else if (adminKey.equals("user"))
+
+		else if (adminKey.equals("user")) {
+			if (getWsContext().getUserPrincipal() == null)
+				throw new IllegalArgumentException("Anon admin is not authenticated, cannot issue any ballot.");
```

Comment:
"The user is not authenticated, cannot issue a ballot without the admin key" would be a better eror message.

Reply:
ok